client/scripts/chat.js

```diff
@@ -1,56 +1,146 @@
     this._chatForm.addEventListener('submit', this._chatSubmitHandler.bind(this));
   }
 
-  _chatSubmitHandler (event) {
+  _handleEmojiSelect(event) {
+    const target = event.target;
+    if (target.dataset.symbol) {
+      const symbol = target.dataset.symbol;
+      this._messageInput.value += symbol;
+    }
+  }
+
+  _chatSubmitHandler(event) {
     event.preventDefault();
     this._handleMessageSubmit(this._messageInput.value);
     this._addSentMessage(this._messageInput.value);
     this._messageInput.value = '';
   }
 
-  updateUsers (users) {
+  updateUsers(users) {
     this._usersList.innerHTML = '';
     users.forEach((username) => {
-      this._addListElement(username, [ 'user' ], this._usersList);
+      this._addListElement(username, ['user'], this._usersList);
+    });
+  }
+
+  _fillEmojiContainer(map) {
+    Object.keys(map).forEach((key) => {
+      const image = this._createImage();
+      image.src = map[key];
+      image.classList.add('emoji__select');
+      image.dataset.symbol = key;
+      this._emojiContainer.appendChild(image);
     });
   }
 
   _addSentMessage(msg) {
-    this._addListElement(msg, ['message', 'message-dispatched'], this._messagesList);
+    msg = this._replaceSymbolsWithEmoji(msg, this._imagesMap);
+    const sentMessage = this._createMessageContainer('', msg);
+    this._addListElement(sentMessage, ['message', 'message-dispatched'], this._messagesList);
     this._scrollToTop(this._messagesList);
   }
 
-  addMessage (msg) {
-    this._addListElement(msg, [ 'message', 'message-received' ], this._messagesList);
+  addMessage(props) {
+    const messageBody = this._replaceSymbolsWithEmoji(props.message, this._imagesMap);
+    const receivedMessage = this._createMessageContainer(props.name, messageBody);
+    this._addListElement(receivedMessage, ['message', 'message-received'], this._messagesList);
     this._scrollToTop(this._messagesList);
   }
 
-  _scrollToTop (container) {
+  _createMessageContainer(name, message) {
+    const container = document.createElement('div');
+    const nameEl = document.createElement('p');
+    const messageBody = document.createElement('p');
+    nameEl.classList.add('user__name');
+    nameEl.innerHTML = name;
+    messageBody.classList.add('message__body');
+    messageBody.innerHTML = message;
+    container.appendChild(nameEl);
+    container.appendChild(messageBody);
+    return container.outerHTML;
+  }
+
+  _mapEmojiToImages(emojiMap) {
+    return Promise.all(Object.keys(emojiMap).map((key) => {
+      return import(`./../images/emoji/${emojiMap[key]}.svg`)
+          .then((src) => {
+            return {
+              symbol: key,
+              src: src
+            };
+          })
+          .catch((e) => {
+            alert(e);
+          });
+    }));
+  }
+
+  _replaceSymbolsWithEmoji(message, emoticons) {
+    let result = message;
+
+    Object.keys(emoticons).map((emojiSymbol) => {
+      if (result.includes(emojiSymbol)) {
```

Comment:
This check is just another iteration over message string. No real need to do it, just proceed with logic